### DISCUSSION TITLE
Unsupervised anomaly detection in heterogeneous graphs?

### DISCUSSION THREAD

**Discussion by monk1337** [pyg-team/pytorch_geometric#6095] [Q&A]
Hi, I am trying to find anomalies in heterogeneous or Knowledge graphs. For example, if I use the famous [MovieLens dataset](https://pytorch-geometric.readthedocs.io/en/latest/notes/load_csv.html) and construct a heterogeneous graph, How do I go about finding anomalies as no ground truth or labels are available?

Guidance with examples would be really helpful.
Thank you!

**Comment by rusty1s** [comment]
I am not really knowledge about anomaly detection, but there exists an interesting library built upon PyG you might take inspiration from: https://github.com/pygod-team/pygod

Other than that, we provide an example of unsupervised learning on heterogeneous graphs in `examples/hetero/dmgi_unsup.py` which you might be able to leverage to find anomalies in output embeddings.
